docker-entrypoint.sh

```diff
@@ -1,61 +1,366 @@
+env | while read -r line; do
+  debug "${line}"
+done
+info "Any config file settings can be overwritten by environment variables."
+info "Use 'CONTAINER_LOG_LEVEL=DEBUG' to see the full list of active environment variables for this container."
+
+echo "###########################"
+echo "# Container sanity checks #"
+echo "###########################"
+info "Running container as user $(whoami) with id $(id -u) and group $(id -g)"
+if [[ $(id -u) -eq 0 ]] || [[ $(id -g) -eq 0 ]] ; then
+  debug "User and group are set to root with id 0 and group 0."
+elif [[ $(id -u) -eq 1000 ]] || [[ $(id -g) -eq 1000 ]] ; then
+  debug "User and group are set to 1000 and group 1000."
+else
+  # Test if the user tampered with the user and group settings
+  warning "Running container as user '$(whoami)' with id $(id -u) and group $(id -g)"
+  warning "Changing these values is only recommended if you're an advanced docker user and can handle file permission issues yourself."
+  warning "Consider leaving the user and group options as root with 0:0 or 1000:1000 or avoid that setting completely."
+  warning "If you need to change the user and group, make sure to rebuild the docker image with the appropriate UID,GID build args."
+fi
 
 if [[ -d /ors-core ]] || [[ -d /ors-conf ]]; then
-  echo "You're mounting old paths. Remove them and migrate to the new docker setup: https://giscience.github.io/openrouteservice/run-instance/installation/running-with-docker"
-  echo "Exit setup due to old folders /ors-core or /ors-conf being mounted"
-  sleep 5
-  exit 1
+  warning "Found remnants of old docker setup."
+  warning "The ors-core and ors-conf folders are not used by default with the new docker setup."
+  warning "Continuing with the new docker setup."
+fi
+
+# Fail if ORS_HOME env var is not set or if it is empty or set to /
+if [ -z "${ORS_HOME}" ] || [ "${ORS_HOME}" = "/" ]; then
+  critical "ORS_HOME is not set or empty or set to /. This is not allowed. Exiting."
+fi
+
+mkdir -p "${ORS_HOME}" || critical "Could not create ${ORS_HOME}"
+
+# Make sure ORS_HOME is a directory
+if [ ! -d "${ORS_HOME}" ]; then
+  critical "ORS_HOME: ${ORS_HOME} doesn't exist. Exiting."
+elif [ ! -w "${ORS_HOME}" ]; then
+  error "ORS_HOME: ${ORS_HOME} is not writable."
+  error "Make sure the file permission of ${ORS_HOME} in your volume mount are set to $(id -u):$(id -g)."
+  error "Under linux the command for a volume would be: sudo chown -R $(id -u):$(id -g) /path/to/ors/volume"
+  critical "Exiting."
 fi
 
-ors_base=${1}
-catalina_base=${ors_base}/tomcat
-graphs=${ors_base}/ors-core/data/graphs
+success "ORS_HOME: ${ORS_HOME} exists and is writable."
 
-echo "ORS Path: ${ors_base}"
-echo "Catalina Path: ${catalina_base}"
+mkdir -p "${ORS_HOME}"/{files,logs,graphs,elevation_cache,config} || warning "Could not create ${ORS_HOME} and folders"
+mkdir -p "${ORS_HOME}"/{files,logs,graphs,elevation_cache,config} || warning "Could not create ${ORS_HOME} and folders"
```

Comment:
`ORS_HOME` is created above, so this should only report errors on folders.
Also, why is this done twice?

Reply:
adjusted